bundles/org.openhab.automation.jsscripting/src/main/java/org/openhab/automation/jsscripting/internal/OpenhabGraalJSScriptEngine.java

```diff
@@ -381,27 +414,61 @@ public void close() throws Exception {
     }
 
     /**
-     * Tests if the current script is a UI-based script, i.e. it is neither loaded from a file nor a transformation.
+     * Tests if the script is a script file, i.e. it is loaded from a file.
      * 
-     * @return true if the script is UI-based, false otherwise
+     * @return
      */
-    private boolean isUiBasedScript() {
+    private boolean isScriptFile() {
         ScriptContext ctx = delegate.getContext();
         if (ctx == null) {
             logger.warn("Failed to retrieve script context from engine '{}'.", engineIdentifier);
             return false;
         }
-        return ctx.getAttribute("javax.script.filename") == null
-                && !engineIdentifier.startsWith(OPENHAB_TRANSFORMATION_SCRIPT);
+        return ctx.getAttribute("javax.script.filename") != null;
+    }
+
+    /**
+     * Get the module type id (if any) of the module executing the script.
+     * 
+     * @return
+     */
+    private @Nullable String getModuleTypeId() {
+        ScriptContext ctx = delegate.getContext();
+        if (ctx == null) {
+            logger.warn("Failed to retrieve script context from engine '{}'.", engineIdentifier);
+        }
+
+        Object value = ctx.getAttribute(AbstractScriptModuleHandler.CONTEXT_KEY_MODULE_TYPE_ID);
+        if (value instanceof String str) {
+            return str;
+        }
+        return null;
+    }
+
+    /**
+     * Tests if a script is a script action, i.e. executed by the ScriptActionHandler.
+     * 
+     * @return
+     */
+    private boolean isScriptAction() {
+        return ScriptActionHandler.TYPE_ID.equals(getModuleTypeId());
+    }
+
+    /**
+     * Tests if the script is a script condition, i.e. executed by the ScriptConditionHandler.
+     * 
+     * @return
+     */
+    private boolean isScriptCondition() {
+        return ScriptConditionHandler.TYPE_ID.equals(getModuleTypeId());
     }
 
     /**
-     * Tests if the current script is a transformation script, i.e. it is created from the script transformation
-     * service.
+     * Tests if the script is a transformation script, i.e. created from the script transformation service.
      * 
-     * @return true if the script is a transformation script, false otherwise
```

Comment:
It is "normal convention" in JavaDocs to more or less repeat some of the general description in the `@return` description. It might seem "stupid" when read like this, but when read via information lookup, it's useful to be able to read what each parameter and the return value is, without reading the whole general description.